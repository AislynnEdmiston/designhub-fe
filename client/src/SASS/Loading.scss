.Loading-container {
  #svgContainer {
    max-width: 50%;
    max-height: 50%;
    position: absolute;
    top: 0%;
    left: 0;
    right: 0;
    bottom: 0;
    margin: auto;
    opacity: 0.2;
  }

  .loading-text {
    position: absolute;
    width: max-content;
    top: 60%;
    left: 0;
    right: 0;
    margin: 0 auto;
    color: white;
    opacity: 0.2;
    animation: blink 0.8s alternate-reverse infinite ease-in-out;
  }

  @keyframes blink {
    from {
      opacity: 0.1;
    }
    to {
      opacity: 0.2;
    }
  }
}
