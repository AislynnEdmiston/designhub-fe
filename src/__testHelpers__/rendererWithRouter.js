import React from 'react'
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';

import rootReducer from '../store/reducers';
jest.mock('../auth-wrapper');

const store = createStore(
  rootReducer,
  applyMiddleware(thunk)
);
const rendererWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Provider store={store}>
        <Router history={history}>
          {ui}
        </Router>
      </Provider>
    ),
    history
  }
}

export default rendererWithRouter;
