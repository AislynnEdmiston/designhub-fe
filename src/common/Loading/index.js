import React from 'react';
import './styles.scss';

export default function Loading() {
  return (
    <div className="loading-wrapper">
      <h1>Loading...</h1>
    </div>
  );
}
