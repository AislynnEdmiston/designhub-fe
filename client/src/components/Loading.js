import React from 'react';
import bodymovin from 'bodymovin';

import loading from '../ASSETS/loading.json';

import '../SASS/Loading.scss';

const Loading = () => {
  var svgContainer = document.getElementById('svgContainer');
  var animItem = bodymovin.loadAnimation({
    wrapper: svgContainer,
    animType: 'svg',
    loop: true,
    path: 'https://assets5.lottiefiles.com/datafiles/AX0rqrGV5ahKpWr/data.json'
  });
  return (
    <div className="Loading-container">
      <div id="svgContainer"></div>
      <p className="loading-text">Loading awesome content...</p>
    </div>
  );
};

export default Loading;
