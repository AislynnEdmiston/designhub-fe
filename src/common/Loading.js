import React from 'react';
import { GooSpinner } from 'react-spinners-kit';
import './SASS/Loading.scss';

const Loading = () => {
  return (
    <div className="Loading-container">
      <GooSpinner size={80} color="#5557fe" className="loader" />
    </div>
  );
};

export default Loading;
