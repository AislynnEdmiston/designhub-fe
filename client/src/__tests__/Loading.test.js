import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../components/Loading';

import rendererWithRouter from '../__testHelpers__/rendererWithRouter'

describe('<App />', () => {
  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouter(
        <Loading history={{ location: { pathname: '/' } }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('render', () => {
    const wrapper = shallow(<Loading />);

    it('should render Loading div element successfuly', () => {
      expect(wrapper.find('.Loading-container')).toHaveClassName(
        'Loading-container'
      );
      expect(wrapper.find('.Loading-container')).toExist();
    });
    it('should render Loader successfuly', () => {
      expect(wrapper.find('.loading-text')).toExist();
      expect(wrapper.find('.loading-text')).toHaveClassName('loading-text');
    });
  });
});
