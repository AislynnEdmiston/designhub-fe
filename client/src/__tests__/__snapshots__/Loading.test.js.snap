// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="Loading-container"
>
  <div
    aria-busy="true"
    className="loading-text"
    style={Object {}}
  >
    <svg
      aria-label="audio-loading"
      fill="#C0C0C0"
      height={150}
      viewBox="0 0 105 105"
      width={150}
    >
      <circle
        cx="12.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="100ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="300ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="600ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="800ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="400ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="700ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="500ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="200ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;
